mewarpx/mewarpx/utils_store/profileparser.py

```diff
@@ -0,0 +1,150 @@
+import sys
+import re
+import json
+
+
+class FullProfile(object):
+    """
+    Class for parsing, storing and writing the full profile data
+    """
+    def __init__(self, lines):
+        self.lines = lines
+        self.min = 0
+        self.avg = 0
+        self.max = 0
+        self.profile_dicts = []
+
+    def parse_full_profiling_output(self):
+        print("Parsing profiling output...")
+
+        self.exclusive_profiles = []
+        self.inclusive_profiles = []
+
+        # The first 5 lines are not needed for parsing
+        del self.lines[0:5]
+        index = 0
+        count = 0
+        for line in self.lines:
+            if "------" in line:
+                break
+            # these lines are of the form "Function()  ncalls  excl_min  excl_avg   excl_max  max_percent"
+            # with more spaces than are here
+
+            # replace the multiple spaces with a single space
+            line = re.sub("\s+", " ", line)
+            values = line.split(" ")
+
+            # work backwards through the values in order to capture function names with spaces.
+            max_percent = float(values[-1].replace("%", ""))
+            excl_max = float(values[-2])
+            excl_avg = float(values[-3])
+            excl_min = float(values[-4])
+            n_calls = int(values[-5])
+            name = "".join(values[0:-5])
```

Comment:
Could you use a space or underscore for the joining character, rather than an empty string, so that the resulting analysis will have readable output?

Reply:
Ditto below

Reply:
Great idea